test/lib/rules/consistent-component-name.js

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+
+const { RuleTester } = require('eslint');
+
+const { ESLINT_TEST_CONFIG } = require('../shared');
+const rule = require('../../../lib/rules/consistent-component-name');
+
+const ruleTester = new RuleTester(ESLINT_TEST_CONFIG);
+
+ruleTester.run('consistent-class-name', rule, {
+    valid: [
+        {
+            code: `export class Foo extends LightningElement {}`,
+            filename: 'foo.js',
+        },
+        {
+            code: `export class ComplexName extends LightningElement {}`,
+            filename: 'complexName.js',
+        },
+    ],
+    invalid: [
+        {
+            code: `export default class extends LightningElement {}`,
+            filename: 'foo.js',
+            errors: [
+                {
+                    type: 'ClassDeclaration',
+                    column: 16,
+                    message: 'Lightning component class should be named "Foo".',
+                },
+            ],
+            output: 'export default class Foo extends LightningElement {}',
```

Comment:
The `RuleTester` we import from eslint automatically verifies the code fix here, correct? Looking through the source I believe it does but wanted to double check.

Reply:
Yes, it's the case. If the `output` is not the generated output by the fixer the test fails.